theano/gpuarray/opt.py

```diff
@@ -97,6 +95,15 @@ def f(local_opt):
     return f
 
 
+def register_opt2(tracks, *tags, **kwargs):
+    def f(local_opt):
+        name = (kwargs and kwargs.pop('name')) or local_opt.__name__
+        opt = theano.gof.local_optimizer(tracks)(local_opt)
+        gpu_optimizer2.register(name, opt, 'fast_run', 'gpuarray', *tags)
```

Comment:
'fast_run' is there by default. We add 'fast_compile' when needed. We add 'fast_compile' only for opt that lift to the GPU. We don't do it for other opt.